inference-engine/src/mkldnn_plugin/nodes/non_max_suppression.cpp

```diff
@@ -20,82 +22,156 @@ class NonMaxSuppressionImpl: public ExtLayerBase {
+                precCheck(iouThrTensorDesc, "iou_threshold");
+                dimsCheck(iouThrTensorDesc, "iou_threshold");
             }
 
             if (layer->insData.size() > 4) {
-                if (layer->insData[NMS_SCORETHRESHOLD].lock()->getTensorDesc().getPrecision() != Precision::FP32)
-                    THROW_IE_EXCEPTION << layer->name << " Incorrect 'score_threshold' input precision. Only FP32 is supported!";
-                SizeVector score_threshold_dims = layer->insData[NMS_SCORETHRESHOLD].lock()->getTensorDesc().getDims();
-                if (score_threshold_dims.size() && score_threshold_dims[0] != 1)
-                    THROW_IE_EXCEPTION << layer->name << " 'score_threshold' should be scalar";
+                const TensorDesc &scoreThrTensorDesc = layer->insData[NMS_SCORETHRESHOLD].lock()->getTensorDesc();
+                precCheck(scoreThrTensorDesc, "score_threshold");
+                dimsCheck(scoreThrTensorDesc, "score_threshold");
             }
 
-            if (layer->outData[0]->getTensorDesc().getPrecision() != Precision::I32)
-                THROW_IE_EXCEPTION << layer->name << " Incorrect 'selected_indices' input precision. Only I32 is supported!";
-            SizeVector selected_indices_dims = layer->outData[0]->getTensorDesc().getDims();
-            if (selected_indices_dims.size() != 2 || selected_indices_dims[1] != 3)
-                THROW_IE_EXCEPTION << layer->name << " 'selected_indices' should be with shape [num_selected_indices, 3]";
+            if (layer->insData.size() > 5) {
+                const TensorDesc &softNmsSigmaTensorDesc = layer->insData[NMS_SOFTNMSSIGMA].lock()->getTensorDesc();
+                precCheck(softNmsSigmaTensorDesc, "soft_nms_sigma");
+                dimsCheck(softNmsSigmaTensorDesc, "soft_nms_sigma");
+            }
 
-            center_point_box = layer->GetParamAsBool("center_point_box", false);
-            sort_result_descending = layer->GetParamAsBool("sort_result_descending", true);
+            const TensorDesc &selIndTensorDesc = layer->outData[NMS_SELECTEDINDICES]->getTensorDesc();
+            if (selIndTensorDesc.getPrecision() != Precision::I32 && selIndTensorDesc.getPrecision() != Precision::I64)
+                THROW_IE_EXCEPTION << logPrefix << " has unsupported 'selected_indices' output precision: " << selIndTensorDesc.getPrecision();
+            const SizeVector &selected_indices_dims = selIndTensorDesc.getDims();
+            if (selected_indices_dims.size() != 2)
+                THROW_IE_EXCEPTION << logPrefix << "has unsupported 'selected_indices' output rank: " << selected_indices_dims.size();
+            if (selected_indices_dims[1] != 3)
+                THROW_IE_EXCEPTION << logPrefix << "has unsupported 'selected_indices' output 2nd dimension size: " << selected_indices_dims[1];
+
+            if (layer->outData.size() > 1) {
```

Comment:
layer->outData.size() > NMS_SELECTEDSCORES more clear and safely in case of NMS_SELECTEDSCORES modifications.

Reply:
Also please modify all other conditions.

Reply:
done